src/Evolution/Systems/NewtonianEuler/Subcell/ResizeAndComputePrimitives.hpp

```diff
@@ -0,0 +1,73 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/VariablesTag.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/ActiveGrid.hpp"
+#include "Evolution/DgSubcell/Tags/ActiveGrid.hpp"
+#include "Evolution/DgSubcell/Tags/Mesh.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+template <size_t Dim>
+class Mesh;
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace NewtonianEuler::subcell {
+/*!
+ * \brief Mutator that resizes the primitive variables to have the size of the
+ * active mesh and then computes the primitive variables on the active mesh.
+ *
+ * In the DG-subcell `step_actions` list this will normally be called using the
+ * `::Actions::MutateApply` action right after the
+ * `evolution::dg::subcell::Actions::TciAndSwitchToDg` action. In the case the
+ * active grid switches from subcell to DG, the resize and primitives are
+ * compute, while if the active grid remains subcell then the primitives are
+ * computed on the subcells.
```

Comment:
Well this is in direct contradiction to the implementation, which does nothing in the "grid remains subcell" case and explains it with a nice comment. Can the nice comment also go here? :)